geode-wan/src/main/java/org/apache/geode/cache/client/internal/locator/wan/LocatorDiscovery.java

```diff
@@ -133,75 +154,69 @@ public void run() {
   }
 
   private WanLocatorDiscoverer getDiscoverer() {
-    return this.discoverer;
+    return discoverer;
   }
 
   private void exchangeLocalLocators() {
     int retryAttempt = 1;
     while (!getDiscoverer().isStopped()) {
       try {
-        RemoteLocatorJoinResponse response =
-            (RemoteLocatorJoinResponse) locatorClient.requestToServer(locatorId.getHost(),
-                request, WanLocatorDiscoverer.WAN_LOCATOR_CONNECTION_TIMEOUT, true);
+        RemoteLocatorJoinResponse response = (RemoteLocatorJoinResponse) locatorClient
+            .requestToServer(locatorId.getHost(), request, WAN_LOCATOR_CONNECTION_TIMEOUT, true);
         if (response != null) {
-          LocatorHelper.addExchangedLocators(response.getLocators(), this.locatorListener);
+          addExchangedLocators(response);
           logger.info("Locator discovery task exchanged locator information {} with {}: {}.",
```

Comment:
While you're changing this method, can you change this this log message a bit?

Currently it looks like:
```
Locator discovery task exchanged locator information 10.212.3.135[10332] with localhost[10331]: {1=[10.212.3.135[10331]], 2=[10.212.3.135[10332]]}.
```
It would make more sense with something like:
```
Locator discovery task for locator 10.212.3.135[10332] exchanged locator information with localhost[10331]: {1=[10.212.3.135[10331]], 2=[10.212.3.135[10332]]}.
```

Reply:
Done!